### PR TITLE
Attesting Indices Fix

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Conforms the helper to the spec and actually filters for the attesting indices. 
- [x] Fix current tests and add regression tests

### CODE REVIEW

File: beacon-chain/core/helpers/committee_test.go [hunk 1]
```diff
@@ -246,19 +246,19 @@ func TestAttestationParticipants_NoCommitteeCache(t *testing.T) {
...
-			wanted:          []uint64{85, 103},
+			wanted:          []uint64{85},
 		},
 		{
 			attestationSlot: 11,
 			stateSlot:       10,
 			bitfield:        []byte{0x03},
-			wanted:          []uint64{102, 68},
+			wanted:          []uint64{68, 102},
 		},
```
Comment (prestonvanloon) [inline beacon-chain/core/helpers/committee_test.go]: Please add a case for bitfield: []byte{0x00} and wanted empty list.

Comment (nisdas) [reply beacon-chain/core/helpers/committee_test.go]: I added a test for that 
```
TestAttestationParticipants_EmptyBitfield
```

Comment (prestonvanloon) [reply beacon-chain/core/helpers/committee_test.go]: Please add it here too

Comment (prestonvanloon) [reply beacon-chain/core/helpers/committee_test.go]: OK it was merged anyway, i'll add it

**Pull request merged**